### ISSUE TITLE
iOS: Optimistic premium upgrade

### ISSUE THREAD

**Issue by samuelclay** [samuelclay/NewsBlur#1691]
To address the issue of Apple's servers sometimes not sending payment receipts until a few minutes later, let's optimistically upgrade ios users who complete the payment flow successfully. 

We have the `/profile/save_ios_receipt` endpoint, so let's reuse that. Use the same `product_identifier` and use `in-progress` for the transaction_identifier. The backend will take care of duplicates (within 3 days).

**Comment by Dejal** [comment]
I noticed a bug: there was no `break` statement for the `SKPaymentTransactionStateDeferred` case, which meant that it would fall through to the `SKPaymentTransactionStateFailed` case, and thus tell the user that the purchase failed, and never process the deferred transaction. Oops. That's fixed.

But I also added the optimistic logic, so it sends `in-progress` to that endpoint (I could send the transaction identifier, but I assume you want an indicator that it isn't complete).

The backend probably should revert the user to non-premium if the endpoint isn't called again with a transaction identifier within those three days.

If you wish, I could send to that endpoint if the transaction fails, perhaps with a `failed` value for the transaction identifier, so it can clear the premium state at that time. (Or a tidier approach could be to add a `state` parameter, with `success`, `in-progress`, and `failed` values.)

I'll close this, since I've done what was asked for, but I can make further changes if you wish.
